processor/spanmetricsprocessor/cache/cache.go

```diff
@@ -0,0 +1,57 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/spanmetricsprocessor/cache"
+
+import lru "github.com/hashicorp/golang-lru"
+
+// Cache consists of an LRU cache and the evicted items from the LRU cache.
+// This data structure makes sure all the cached items can be retrieved either from the LRU cache or the evictedItems
+// map. In spanmetricsprocessor's use case, we need to hold all the items during the current processing step for
+// building the metrics. The evicted items can/should be safely removed once the metrics are built from the current
+// batch of spans.
+type Cache struct {
+	*lru.Cache
+	evictedItems map[interface{}]interface{}
+}
+
+// NewCache creates a Cache.
+func NewCache(size int) (*Cache, error) {
```

Comment:
Is there a reason that we are only replacing the `metricKeyToDimensions` map? Since all the maps have unbounded memory usage, why just this one?

Reply:
Would you expect this cache to be reset in the case of delta metrics?

Reply:
> Is there a reason that we are only replacing the metricKeyToDimensions map? Since all the maps have unbounded memory usage, why just this one?

For the other maps, we are not caching but aggregating. For delta metrics, we should reset the maps every time(as you implemented in the other PR). For cumulative metrics, we can't reset it. That's why we said this PR could only partially addresses the issue: #2179.

> Would you expect this cache to be reset in the case of delta metrics?
It's safe to reset this map every time regardless of delta or cumulative metrics. However, we should not do that for performance consideration. The cache is used for speeding up resource attributes lookup when we build the metrics.